common/src/View/MapFrame.cpp

```diff
@@ -654,7 +654,7 @@ namespace TrenchBroom {
 
         void MapFrame::gridDidChange() {
             const Grid& grid = m_document->grid();
-            m_gridChoice->SetSelection(static_cast<int>(grid.size()));
+            m_gridChoice->SetSelection(grid.size() - Grid::MinSize);
```

Comment:
Maybe spend a method that computes the index instead of repeating the computation, e.g. `getGridSizeIndex()` or something.